### PR TITLE
Server refactor+testing+renaming

### PR DESCRIPTION
**Pull request by plauche**
Ok so maybe there was some scope creep in this refactor...

### Added

- Added CI workflows for running clippy and tests.

### Changed

- Renamed `block-streamer` to `myceli` and `server` to `listener`.
- Reorganized the functionality in `block_streamer::server` to make functional testing much easier/possible.
- Small refactor of `transmit_blocks`.

### Removed

- Removed the `receive` mode from `block-streamer/myceli`, as it was already covered by the `listener/server` functionality.

### CODE REVIEW

**Comment by plauche** [comment]
I'm not sure why all of the `myceli` source appears to be new because it's definitely just moved from `block-streamer`...

**Comment by ddimaria** [comment]
@plauche are you adding some basic CI to this branch (lints/tests)?

File: myceli/src/handlers.rs [hunk 1]
```diff
@@ -0,0 +1,247 @@
+use crate::transmit::transmit_blocks;
```
Comment (plauche) [inline myceli/src/handlers.rs:1]: @ddimaria this file contains the most "new" code and actually new tests

File: myceli/src/listener.rs [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+                }
+                Err(err) => {
+                    error!("Message parse failed: {err}");
+                }
+            }
+        }
+    }
+
+    async fn handle_message(&mut self, message: Message) -> Result<Option<Message>> {
+        let resp = match message {
```
Comment (plauche) [inline myceli/src/listener.rs:76]: This new "passing responses back from the message handlers" logic is a nice cleanup, but I'm not completely satisfied with it yet. Some messages will never have a response that is handled this way, so I may eventually split this match into one with handlers with responses-to-be-sent, and one with handlers without those responses.

File: .github/workflows/linting.yaml [hunk 3]
```diff
@@ -0,0 +1,17 @@
+name: CI Linting
+on: push
```
Comment (ddimaria) [inline .github/workflows/linting.yaml:2]: Limit to PRs, or just keep open for any push?

File: CHANGELOG.md [hunk 4]
```diff
@@ -10,12 +10,15 @@
 
 ### Changed
 
+- Renamed `block-streamer` to `myceli` and `server` to `listener`.
```
Comment (ddimaria) [inline CHANGELOG.md:13]: I just saw the latest episode last night, great show 💯

**Review by ddimaria — APPROVED** [review]

File: .github/workflows/linting.yaml (hunk 3 above)
Comment (plauche) [reply .github/workflows/linting.yaml]: Good catch! Made that specific to PRS.

**Pull request merged**